dv/uvm/core_ibex/directed_tests/encoding.h

```diff
@@ -0,0 +1,2876 @@
+/* See LICENSE for license details. */
```

Comment:
Where did this come from? It looks derived from either https://github.com/riscv-non-isa/riscv-arch-test/blob/main/riscv-test-suite/env/encoding.h, this earlier version of it looks closer: https://github.com/riscv/riscv-test-env/blob/0666378f353599d01fc48562b431b1dd049faab5/encoding.h but still not identical.

As this isn't a file authored by us we should at least ensure there's a proper header providing attribution (the current comment refers to a 'LICENSE' file but the top-level LICENSE does not apply to this). Preferably we'd vendor this in rather than adding it directly here. As I think we're vendoring in riscv-arch-test anyway to support the architectural test suite maybe we should just use the `encoding.h` from there?

Reply:
It was from a submodule of riscv-tests ([riscv-test-env](https://github.com/riscv/riscv-test-env/tree/master)). Now, I added a separate file ibex_macros.h for implementation specific stuff while using the encoding.h from the vendored repo without any modification.